src/Forms/Traits/MultipleOptionsTrait.php

```diff
@@ -104,6 +105,42 @@ public function fromQuery(Connection $pdo, $sql, $data = array(), $groupBy = fal
         return $this->fromResults($results, $groupBy);
     }
 
+    /**
+     * Build options array from a DataSet, as provided by <b>Domain</b> gateways
+     * 
+     * @param Dataset $dataset
+     * @param string $valCol
+     * @param string $nameCol
+     * @param string $groupBy
+     * 
+     * @return self
+     */
+    public function fromDataSet(Dataset $dataset, $valCol, $nameCol, $groupBy = false)
+    {
+        if(empty($dataset) || !is_object($dataset)){
```

Comment:
I think in this case the type-hint for `Dataset` in the method signature will already throw an exception if the argument is not an object of the correct type.